libbpf-rs/src/streams.rs

```diff
@@ -0,0 +1,105 @@
+use core::ffi::c_void;
+use std::io;
+
+use crate::libbpf_sys::bpf_prog_stream_read_opts;
+
+use crate::Error;
+use crate::Result;
+
+/// [`Stream`] instances.
+///
+/// `streams`s are BPF descriptors that provide a character output
+/// interface from the BPF program to userspace.
+#[derive(Debug)]
+pub struct Stream {
+    fd: i32,
+    stream_id: u32,
+    bytes_per_read: usize,
+}
+
+/// Default BPF stdout stream id.
+pub const BPF_STDOUT: u32 = 1;
+/// Default BPF stderr stream id.
+pub const BPF_STDERR: u32 = 2;
+
+/// Config options, currently empty and here for forward compatibility.
+struct StreamReadOpts {
+    sz: usize,
+}
+
+impl Stream {
+    /// Default read batch for when the user doesn't specify one.
+    const BYTES_PER_READ_DFL: usize = 4096;
+
+    /// Create a new Stream instance.
+    pub fn new(fd: i32, stream_id: u32) -> Self {
+        Stream {
+            fd,
+            stream_id,
+            bytes_per_read: Self::BYTES_PER_READ_DFL,
+        }
+    }
+
+    /// Specify the maximum read size for stream reads.
+    pub fn new_with_capacity(fd: i32, stream_id: u32, bytes_per_read: usize) -> Self {
+        Stream {
+            fd,
+            stream_id,
+            bytes_per_read,
+        }
+    }
+
+    /// Get a default `StreamReadOpts` struct, since currently the struct
+    /// has no actual data and requiring the user to provide it is
+    /// unnecessarily complicated..
+    fn opts_dfl() -> StreamReadOpts {
+        StreamReadOpts { sz: 0 }
+    }
```

Comment:
This constructor should probably be replaced by a [`Default`](https://doc.rust-lang.org/std/default/trait.Default.html) impl?